include/cudaq/Optimizer/CodeGen/Peephole.h

```diff
@@ -55,6 +55,17 @@ inline mlir::Value createMeasureCall(mlir::PatternRewriter &builder,
   return {};
 }
 
+inline mlir::Value createReadResultCall(mlir::PatternRewriter &builder,
+                                        mlir::Location loc,
+                                        mlir::OpResult result) {
+  auto i1Ty = mlir::IntegerType::get(builder.getContext(), 1);
+  return builder
+      .create<mlir::LLVM::CallOp>(loc, mlir::TypeRange{i1Ty},
+                                  cudaq::opt::QIRReadResultBody,
+                                  mlir::ArrayRef<mlir::Value>{result})
+      .getResult();
```

Comment:
Is the `getResult()` needed?

Reply:
This piece of code was in [0.4.0](https://github.com/NVIDIA/cuda-quantum/blob/releases/v0.4.0/include/cudaq/Optimizer/CodeGen/Peephole.h#L66), and I am just adding it back in. That being said, I tried removing it just for testing purposes, and it failed to compile. So I tried changing this function's return type to `void` and not returning the `builder.create()` result, but that also fails to compile (with a different error). I assume that the tablegen format really wants it to return an `mlir::Value` here, but let me know if I'm missing something.